src/control/cmd/daos_server/storage_scm.go

```diff
@@ -35,11 +41,61 @@ type scmCmd struct {
 type prepareSCMCmd struct {
 	cmdutil.LogCmd `json:"-"`
 	helperLogCmd   `json:"-"`
+	cfgCmd         `json:"-"`
+	socketCmd      `json:"-"`
 
 	NrNamespacesPerSocket uint `short:"S" long:"scm-ns-per-socket" description:"Number of PMem namespaces to create per socket" default:"1"`
 	Force                 bool `short:"f" long:"force" description:"Perform SCM operations without waiting for confirmation"`
 }
 
+func setSockFromCfg(log logging.Logger, cfg *config.Server, affSrc config.EngineAffinityFn, req *storage.ScmPrepareRequest) error {
+	msgSkip := "so skip fetching sockid from config"
+
+	if cfg == nil {
+		log.Debug("nil input server config " + msgSkip)
+		return nil
+	}
+	if req.SocketID != nil {
+		log.Debug("sockid set in req " + msgSkip)
+		return nil
+	}
+
+	// Set engine NUMA affinities accurately in config.
+	if err := cfg.SetEngineAffinities(log, affSrc); err != nil {
+		log.Error(fmt.Sprintf("failed to set engine affinities %s: %s", msgSkip, err))
+		return nil
```

Comment:
We don't want to return an error in this case?

Reply:
I didn't want to block users from manually specifying which PMem to prepare in the case that for some reason the affinity source function fails. do you think that's okay?

Reply:
That makes sense, but isn't that case covered by the `req.SocketID != nil` check above? I don't see this as being a huge issue, but I'm just trying to understand the error cases.

Reply:
yes it is indeed, thank you I changed it to return a failure, done